File: src/main/webapp/app/js/controller/wqtController.js
wqtControllers.controller('wqtController', function($scope, wqtService) {

	wqtService.getContactInformation({}, function(data) {
		$scope.contactInformation = data;
	});

	wqtService.getTicket({
		type : "public"
	}, function(data) {
		$scope.currentTicket = data;
	});

	wqtService.getTicket({
		type : "private"
	}, function(data) {
		$scope.myTicket = data;
		if ($scope.myTicket.arrival !== undefined && $scope.myTicket.arrival !== null) {
			$scope.pickUpPeriod = new Date(new Date().getTime() - new Date($scope.myTicket.arrival).getTime());
		}
	});

	wqtService.getTickets({}, function(data) {
		$scope.tickets = data;
	});

	wqtService.checkTickets();

	$scope.createTicket = function() {
		wqtService.createTicket(function(data) {
			$scope.myTicket = data;
		});
	};

	$scope.updateTicket = function() {
		wqtService.updateTicket(function(data) {
			$scope.tickets = data;
		});
	}

	$scope.startTicket = function() {
		wqtService.startTicket(function(data) {
			$scope.tickets = data;
		});
	}
});

File: src/main/java/de/cw/wqt/controller/WqtController.java
package de.cw.wqt.controller;

import java.util.List;

import javax.servlet.http.HttpServletRequest;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;

import de.cw.wqt.command.get.GetContactInformationCommand;
import de.cw.wqt.command.get.GetTicketCommand;
import de.cw.wqt.command.get.mapper.GetContactInformationCommandMapper;
import de.cw.wqt.command.get.mapper.GetTicketCommandMapper;
import de.cw.wqt.domain.TicketType;
import de.cw.wqt.service.UserService;
import de.cw.wqt.service.WqtService;

@Controller
@RequestMapping(value = "wqt")
public class WqtController {

    @Autowired
    private WqtService wqtService;

    @RequestMapping(value = "getTicket", method = RequestMethod.GET)
    @ResponseBody
    public GetTicketCommand getTicket(@RequestParam String type, HttpServletRequest request) {
        return GetTicketCommandMapper.map(wqtService.getTicket(TicketType.convert(type), UserService.readUserFromSession(request.getSession())));
    }

    @RequestMapping(value = "createTicket", method = RequestMethod.POST)
    @ResponseBody
    public GetTicketCommand createTicket(HttpServletRequest request) {
        return GetTicketCommandMapper.map(wqtService.createTicket(UserService.readUserFromSession(request.getSession())));
    }

    @RequestMapping(value = "updateTicket", method = RequestMethod.PUT)
    @ResponseBody
    public List<GetTicketCommand> updateTicket() {
        return GetTicketCommandMapper.map(wqtService.updateTicket());
    }

    @RequestMapping(value = "startTicket", method = RequestMethod.PUT)
    @ResponseBody
    public List<GetTicketCommand> startTicket() {
        return GetTicketCommandMapper.map(wqtService.startTicket());
    }

    @RequestMapping(value = "checkTickets", method = RequestMethod.PATCH)
    @ResponseBody
    public void checkTickets() {
        wqtService.checkTickets();
    }

    @RequestMapping(value = "tickets", method = RequestMethod.GET)
    @ResponseBody
    public List<GetTicketCommand> getTickets() {
        return GetTicketCommandMapper.map(wqtService.getTickets());
    }

    @RequestMapping(value = "contact_information", method = RequestMethod.GET)
    @ResponseBody
    public GetContactInformationCommand getContactInformation() {
        return GetContactInformationCommandMapper.map(wqtService.getContactInformation());
    }

}
